### ISSUE TITLE
Initial Conditions

### ISSUE THREAD

**Issue by lindsayad** [aurora-multiphysics/platypus#59]
AFAICT platypus doesn't currently support an `ICs` block. It looks like the MFEM code for creating initial conditions is pretty straightforward (as I'm sure you guys know). This is what INL's AI tool gave me
```c++
    // The initial solution GridFunction
    GridFunction u(&fespace);

    // Define the initial condition coefficient
    FunctionCoefficient u0(InitialCondition);

    // Project the initial condition onto the GridFunction
    u.ProjectCoefficient(u0);
```
where `InitialCondition` is a function taking `Vector` as an input argument. Are there any plans to add support for `ICs`?

**Comment by lindsayad** [comment]
I am curious why this line in `PlatypusApp.C`:
```c++
  addTaskDependency("add_mfem_coefficients", "add_ic");
```
I was thinking that I would actually create an IC that takes a coefficient and then the IC does the coefficient projection onto the grid function, but then the dependency ordering would have to be reversed

**Comment by alexanderianblair** [comment]
Yup - we have it on our task list, but I don't think anyone's had the chance to pick it up yet! The suggested approach (creating a Coefficient and then calling `ProjectCoefficient`) is exactly the route to take.

The task dependency looks vestigial to me, from a forerunner to Platypus that had `mfem::GridFunctionCoefficient` set up in an awkward way - can (and should) be reversed for general ICs here.
